### DISCUSSION TITLE
RocketFuel Theme 3.5.4 released

### DISCUSSION THREAD

**Discussion by leigh-pointer** [oqtane/oqtane.framework#2866] [Show and tell]
Theme upgraded to Bootstrap 5.3
Added Light and Dark Theme support
small bug fixes.

![ThemeChanger](https://github.com/oqtane/oqtane.framework/assets/9447612/2b04e455-159f-449f-97dd-e61c669f50ac)

Enjoy!

**Comment by sbwalker** [comment]
@leigh-pointer should we upgrade the framework to Bootstrap 5.3 for the Oqtane 4.0.0 release? Are there any breaking changes in Bootstrap 5.3?

**Reply by leigh-pointer** [reply]
@sbwalker we can still upgrade bootstrap to 5.3 without any strange effects.

**Comment by leigh-pointer** [comment]
no breaking changes as far as I see.  I will add a PR with an update.  If you want the Light and Dark switch, let me know ?

**Comment by sbwalker** [comment]
@leigh-pointer 

Does the light/dark switch work for every installation regardless of the theme being used or the CSS framework being used in a theme (ie. some themes will not use Bootstrap)? 
How does the light/dark switch behave when the screen size is reduced to a mobile view - does it maintain its responsive appearance?

**Reply by leigh-pointer** [reply]
@sbwalker this is dependent on the theme being Bootstrap based.

**Comment by leigh-pointer** [comment]
@sbwalker the Light/Dark is controlled by an attribute in the <Head> tag. I have just had a look at the bootswatch theme "cyborg" https://bootswatch.com/cyborg/ and it appears not to behave as one would expect.  This issue would need work or a change to the default theme, which I feel is possible keeping the dark theme as default.

There is no change in behavior when the screen size is changed and it maintains the appearance.

**Comment by sbwalker** [comment]
@leigh-pointer we could skip the light/dark option and just upgrade to BS 5.3 - correct?

**Reply by leigh-pointer** [reply]
Yes! I will upgrade Oqtane to 5.3

**Comment by sbwalker** [comment]
@leigh-pointer I took care of this in #2891

**Comment by sbwalker** [comment]
@leigh-pointer one item I have noticed with the upgrade to Bootstrap 5.3 is that it affected the display of tables. The Pager component obviously displays tabular information and on Bootstrap 5.3 the table is now a different color than the page background. I also noticed on components which use a table-based layout the colors are affected as well. Obviously Oqtane moved away from table-based layouts in components a long time ago... but some legacy modules were never upgraded to the modern div method. I am not sure if these behaviors are part of Bootstrap 5.3 or Bootswatch (ie. the Cyborg theme). Either way, I do not consider them to be "breaking" changes as Oqtane should not be held back by legacy markup approaches.

**Comment by sbwalker** [comment]
@leigh-pointer I was able to figure this out... at least for the legacy table-based layouts. This was actually related to a change in Bootstrap 5.0 originally where tables were styled differently than previously. In order to override this behavior, the Oqtane default theme included some styles to preserve the old behavior (Theme.css):

```
.table .form-control {
    background-color: #ffffff !important;
    border-width: 0.5px !important;
    border-bottom-color: #ccc !important;
}

.table .form-select {
    background-color: #ffffff !important;
    border-width: 0.5px !important;
    border-bottom-color: #ccc !important;
}
```

However in Bootstrap 5.3 another change was introduced related to color:

```
color: var(--bs-table-color-state,var(--bs-table-color-type,var(--bs-table-color)));
```

And the result is that the text within input elements within a table is now displayed in a white color (making it invisible). One way to resolve this is to include an overriding color style:

```
.table .form-control {
    background-color: #ffffff !important;
    border-width: 0.5px !important;
    border-bottom-color: #ccc !important;
    color: #000 !important;
}

.table .form-select {
    background-color: #ffffff !important;
    border-width: 0.5px !important;
    border-bottom-color: #ccc !important;
    color: #000 !important;
}
```
Obviously the better solution is for modules to use modern div-based layouts which will be displayed correctly and also support responsive rendering.

**Comment by sbwalker** [comment]
#2937 resolves the cosmetic issue
